src/com/google/javascript/jscomp/AbstractCommandLineRunner.java

```diff
@@ -2009,6 +2011,18 @@ void printManifestTo(Iterable<CompilerInput> inputs, Appendable out)
   void printBundleTo(Iterable<CompilerInput> inputs, Appendable out)
       throws IOException {
 
+    // First, print all defines passed in
+    Set<Entry<String, Node>> defines = compiler.getOptions().getDefineReplacements().entrySet();
+    if (!defines.isEmpty()) {
+      Node[] defineNodes = defines.stream()
```

Comment:
We cannot yet use the `.stream()` method or arrow functions.
These are java 8 features and we currently have limitations that prevent us from using them in most of our code.

Reply:
Sorry about that, I'll back it down - the pom's seem to indicate java8, since otherwise my IDE should not have allowed me to do this...

Reply:
Sorry the poms misled you.

We're already using the java 8 compiler, but we have google-specific limitations that prevent us using java 8 features in the code. I'm working on fixing that, but it will be awhile yet.

Reply:
Makes sense. I'll get all of these updated as soon as I am able, thanks for you and your team's quick replies.

Reply:
FYI, Java 8 can now be used for code in `javascript/jscomp`